limacharlie/Extensions.py

```diff
@@ -57,7 +42,72 @@ def delete( self, extName ):
     
     def getSchema( self, extName ):
         return self._manager._apiCall( 'extension/schema/%s' % ( extName, ), GET )
+    
+    def migrate( self, extName ):
+        return self._manager._apiCall( 'extension/migrate/%s' % ( extName, ), POST, {
+            'oid' : self._manager._oid,
+        } )
 
+    def request( self, extName, action, data = {}, isImpersonated = False ):
+        req = {
+            'oid' : self._manager._oid,
+            'action' : action,
+            'data' : json.dumps( data ),
+        }
+        if isImpersonated:
+            if self._manager._jwt is None:
+                self._manager._refreshJWT()
+            req[ 'impersonator_jwt' ] = self._manager._jwt
+        return self._manager._apiCall( 'extension/request/%s' % ( extName, ), POST, req )
+    
+    def convert_rules(self, extName, isDryRun = True): 
+        extList = self.list()
+        if extName not in extList:
+            return f'ERROR: unable to convert rules. {self._manager._oid} is not subscribed to {extName}'
+        gen_hive = Hive.Hive(self._manager, "dr-general")
```

Comment:
Hive has no attribute Hive this should be Hive(self._manager, "dr-general")

Also did you have a chance to do a live run against some test data?

Reply:
I did, yes. All test in cli and ui work. That just ended up being a naming convention from the import (line 2)
`from limacharlie import Hive`
it's grabbing attributes 'Hive' and 'HiveRecord' from the file 'Hive'

Reply:
So went back to try the individual imports and got a 'circular import' error:
`ImportError: cannot import name 'HiveRecord' from partially initialized module 'limacharlie' (most likely due to a circular import) (/home/adele/python-limacharlie/limacharlie/__init__.py)`

Happy to address this in another way if you have an alternative suggestion but `Hive.Hive` and `Hive.HiveRecord` seems to resolve circular import issue and makes the whole thing work 🤷